### DISCUSSION TITLE
Protection against malicious or fake software in the repository, update only store apps.

### DISCUSSION THREAD

**Discussion by m-fessler** [microsoft/winget-cli#2534] [General]
Hello,

If I understand correctly, after an initial manual review, further updates are mostly automated.
Is there a mechanism that protects against someone suddenly distributing a malicious update after the initial clean software?

Is there a function that allows all store apps (and only these) to be updated without specifying each app individually?
That would be a good workaround, since the automatic update function in the microsoft store unfortunately doesn't always work and can't be scheduled.

Thanks and greetings,
Martin

**Comment by mdanish-kh** [comment]
> If I understand correctly, after an initial manual review, further updates are mostly automated.

That's certainly not the case. Every manifest has to go through manual approval by a [community moderator](https://github.com/microsoft/winget-pkgs/blob/master/Moderation.md) before it can be added to `winget`. You can see it here in the [Windows Package Manager Community Repo](https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+is%3Aclosed) where manifests are submitted.

> Is there a mechanism that protects against someone suddenly distributing a malicious update after the initial clean software?

A detailed response by a community moderator: https://github.com/microsoft/winget-pkgs/discussions/31547#discussioncomment-1530984
To answer your question:

*"I don't see where security risks would be an issue here because every installer goes through Dynamic Analysis (Virus Scan) in the Pipelines' VMs, and if there's a PUA or malware in the installer, it's immediately flagged by the pipelines. The PR is also manually validated by Moderators, in either VMs, or Bare Metal - so installers are always double checked to make sure that it isn't a malicious package intended to steal people's passwords or monitor what they're typing on their keyboard.
Even if the pipelines cannot catch the malware issue, depending on the antivirus software someone has, all installers from WinGet are downloaded to %TEMP%\WinGet, except for .appx(bundle) and .msix(bundle), where your antivirus software will probably scan it before it's executed to install it onto your PC."*

> Is there a function that allows all store apps (and only these) to be updated without specifying each app individually?
That would be a good workaround, since the automatic update function in the microsoft store unfortunately doesn't always work and can't be scheduled.

https://github.com/microsoft/winget-cli/issues/2546#issuecomment-1256269869
As pointed out by a moderator, winget does not have the ability to differentiate between apps installed through MSStore or winget, so I do not believe this functionality exists yet.
